### PR TITLE
Fix N5Store dtype wrong behavior 

### PR DESCRIPTION
**Pull request by mzouink**
[Description of PR]
Fixing issue: https://github.com/zarr-developers/zarr-python/issues/1339

@d-v-b 

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
Launched the workflows. Thanks, @mzouink!

File: zarr/tests/test_n5.py [hunk 1]
```diff
@@ -35,3 +39,17 @@ def test_partial_chunk_decode(chunk_shape: Tuple[int, ...]):
...
     assert np.array_equal(codec_wrapped.decode(codec_wrapped.encode(subchunk)), chunk)
+
+
+def test_dtype_decode():
+    store = 'data/array.n5'
+    n5_store = N5FSStore(store)
+    z = zarr.open(n5_store)
+    z.create_dataset("test", shape=100, dtype="u1")
+    dtype_n5 = json.loads(n5_store["test/.zarray"])["dtype"]
+    shutil.rmtree(store)
```
Comment (d-v-b) [inline zarr/tests/test_n5.py:50]: consider using `atexit.register` here instead, e.g. https://github.com/zarr-developers/zarr-python/blob/4dc6f1f5046708648b5e6b82d0a64a24d1a40566/zarr/tests/test_storage.py#L992

**Comment by d-v-b** [comment]
I feel like this case should be caught by existing tests. I will look through the test code later and see why this isn't happening.

File: zarr/tests/test_n5.py (hunk 1 above)
Comment (mzouink) [reply zarr/tests/test_n5.py]: Thank you! i am using it now

**Comment by mzouink** [comment]
@d-v-b  
The test is feeling in https://github.com/zarr-developers/zarr-python/blob/4dc6f1f5046708648b5e6b82d0a64a24d1a40566/zarr/tests/test_core.py#L1734-L1741

And it is hardcoded + overwritten for N5. I don't think it is the right way to do it

**Comment by joshmoore** [comment]
Re-launched the workflows. Just a heads up, @mzouink, @d-v-b did a good deal of cleaning of test_core.py in https://github.com/zarr-developers/zarr-python/pull/1305/files#diff-acf894bdb70842660ec996213f7ff01514172d0fd4208cbc33a942929e1c7cb2 in case you want to compare there.

File: zarr/tests/test_n5.py [hunk 2]
```diff
@@ -35,3 +39,14 @@ def test_partial_chunk_decode(chunk_shape: Tuple[int, ...]):
     chunk[subslices] = 1
     subchunk = np.ascontiguousarray(chunk[subslices])
     assert np.array_equal(codec_wrapped.decode(codec_wrapped.encode(subchunk)), chunk)
+
+
+def test_dtype_decode():
```
Comment (joshmoore) [inline zarr/tests/test_n5.py:44]: This test will need the fsspec check:

```
@pytest.mark.skipif(have_fsspec is False, reason="needs fsspec")
```

Comment (mzouink) [reply zarr/tests/test_n5.py:46]: Thank you! i didn't know how to do it. i added it.

Comment (joshmoore) [reply zarr/tests/test_n5.py:46]: Cheers!

**Comment by joshmoore** [comment]
The test that was failing is now passing. Thanks, @mzouink. @d-v-b, do you want to sign off on this?

**Comment by mzouink** [comment]
yes @joshmoore. i think it is ready now. thank you for your time.

**Comment by d-v-b** [comment]
looks good to me

**Comment by joshmoore** [comment]
Great, thanks all. I'll get this into the next release. @mzouink, please feel free to add your contribution to docs/release.rst if you would like.

**Comment by joshmoore** [comment]
:tada:

**Pull request merged**